base/path.jl

```diff
@@ -34,8 +34,25 @@ elseif Sys.iswindows()
     const path_dir_splitter = r"^(.*?)([/\\]+)([^/\\]*)$"
     const path_ext_splitter = r"^((?:.*[/\\])?(?:\.|[^/\\\.])[^/\\]*?)(\.[^/\\\.]*|)$"
 
+    const splitdrive_re = let
+        # Slash in either direction.
+        S = raw"[\\/]"
+        # Not a slash in either direction.
+        N = raw"[^\\/]"
```

Comment:
Would it be helpful to make this be a specification of all invalid path characters?
```suggestion
        N = raw"[^\/?*<>:\"|*]"
```

Reply:
It would certainly make it more accurate. I don't have insight whether there would be any drawback to that. (But this particular suggestion misses the backslash and repeats the asterisk.)

Reply:
ah. I figured it would reduce the ordering dependence (at least with ? in there)

more than one drive letter is an illegal path, so not particularly important either way

Reply:
Yes, it would solve the ordering issue. I'll add a commit with more correctness and we can see whether someone sees a problem with that.